specification/esql/_types/TableValuesType.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { double, integer, long } from '@_types/Numeric'
+
+export interface TableValuesInteger {
+  integer: TableValuesIntegerValue[]
+}
+type TableValuesIntegerValue = integer | integer[]
+export interface TableValuesKeyword {
+  keyword: TableValuesKeywordValue[]
+}
+type TableValuesKeywordValue = string | string[]
+export interface TableValuesLong {
+  long: TableValuesLongValue[]
+}
+type TableValuesLongValue = long | long[]
+export interface TableValuesDouble {
+  double: TableValuesLongDouble[]
+}
+type TableValuesLongDouble = double | double[]
```

Comment:
Since these types are mutually exclusive, we should consider modelling them as a `container` like:

```ts
/** @variants container */
export interface TableValues
{
  integer: integer | integer[]
  keyword: string | string[]
  long: long | long[]
  double: double | double[]
}

Reply:
Would that work with your code-generators @Anaethelion @l-trotta ?

Reply:
nit: for it to be a proper container all the fields should be optional.

To explain further the `@container` was designed primarily for `query` in search where the fields you set defines the type of query you use. Thus each [QueryContainer](https://github.com/elastic/elasticsearch-specification/blob/f737f34e18daa7626ac141171a005ca28372dada/specification/_types/query_dsl/abstractions.ts#L107) can only have one field populated. [link to modeling guide](https://github.com/elastic/elasticsearch-specification/blob/main/docs/modeling-guide.md#container)

I think it fits the intent.

Reply:
@nik9000 

> I did it this way because you can't specify more than one key in the object. Is that what @container does?

Yes, the keys inside the container are mutually exclusive. I updated the above code with the missing optional `?` modifier.

Reply:
Oh sorry, I meant the code with the container example in the above comment.

Reply:
Cool. I've pushed some updates.